test/devise/async/model_test.rb

```diff
@@ -3,59 +3,79 @@
+          @admin2.save # after_commit will be invoked 
+          Worker.expects(:enqueue).with(:confirmation_instructions, "Admin", @admin2.id.to_s, instance_of(String), {})
+        end
+        Worker.expects(:enqueue).never 
       end
-
+      
+      # 3. As data changes, mails are not sent till after_commit is invoked 
       it "accumulates notifications to be sent after commit when Model has been changed" do
-        admin = create_admin
         Admin.transaction do
+          admin = create_admin
+
           admin[:username] = "changed_username"
           admin.send_confirmation_instructions
 
-          mailers = admin.send(:devise_pending_notifications) # [:confirmation_instructions, ["RUQUib67wLcCiEyZMwfx", {}]]
-          mailers.size.must_equal 1
-
-          mailer = mailers.first
-          mailer.size.must_equal 2
-          mailer.first.must_equal :confirmation_instructions
-          mailer.last.must_be_instance_of Array
+          mailers = admin.send(:devise_pending_notifications) # [[:confirmation_instructions, ["RUQUib67wLcCiEyZMwfx", {}]]] 
+          mailers = mailers.first           # [:confirmation_instructions, ["RUQUib67wLcCiEyZMwfx", {}]]
+          notification_type = mailers.first # :confirmation_instructions, ["RUQUib67wLcCiEyZMwfx", {}]]
 
-          Worker.expects(:enqueue).never # after_commit will not fire without save
+          mailers.size.must_equal 2         
+          notification_type.must_equal :confirmation_instructions
+          mailers.last.must_be_instance_of Array
+          Worker.expects(:enqueue).with(:confirmation_instructions, "Admin", admin.id.to_s, instance_of(String), {})
         end
       end
 
+      # 4. As data changed & saved, i.e after_commit is invoked hence notifications are queued to worker
       it "triggers the enqueued notifications on save" do
-        admin = create_admin
         Admin.transaction do
+          admin = create_admin
+
+          Worker.expects(:enqueue).never # nothing queued till save 
           admin[:username] = "changed_username"
           admin.send_confirmation_instructions
 
-          mailers = admin.send(:devise_pending_notifications) # [:confirmation_instructions, ["RUQUib67wLcCiEyZMwfx", {}]]
-          mailers.size.must_equal 1
+          mailers = admin.send(:devise_pending_notifications) # [[:confirmation_instructions, ["RUQUib67wLcCiEyZMwfx", {}]]] 
+          mailers = mailers.first           # [:confirmation_instructions, ["RUQUib67wLcCiEyZMwfx", {}]]
+          mailers.size.must_equal 2         
 
-          mailer = mailers.first
-          mailer.size.must_equal 2
-          mailer.first.must_equal :confirmation_instructions
-          mailer.last.must_be_instance_of Array
-
-          admin.save
-          Worker.expects(:enqueue).with(:confirmation_instructions, "Admin", admin.id.to_s, instance_of(String), {})
+          admin.save                     # after_commit will be invoked
+          2.times do
```

Comment:
This construct looks weird. Why haven't you used something like this:
```ruby
Worker.expects(:enqueue).with(:confirmation_instructions, "Admin", admin.id.to_s, instance_of(String), {})
admin.save
```
?

I am not that used to minitest. But I thought that you first set the expectation and then run the code that triggers it. If the code triggers it before the end of the example the expectation is being satisfied. Is this how minitest works here?